blogchain/README.md

```diff
@@ -15,30 +15,204 @@ Welcome to Blogchain: the decentralized writer's blog. Transform your blogs into
+    title: data.title,
+    description: postContent,
+    extra: "blogpost",
+  },
+  contractAddress: data.contract,
+  ownerId: activeAccountId,
+});
 ```
-pnpm install
+
+We can then execute the mint nft method
+
+```typescript
+  await execute({ wallet }, mintCall);
 ```
-and 
-run the project
+
+#### Note: We populate the 'extra' field with the value 'blogpost' to subsequently filter the displayed NFTs and blogs in blogchain, ensuring that only blogs are included.
+
+-----
+
+## Get Data
+
+#### Get blog posts (nfts) from a blog (smart contract)
+
+Using [mintbase GraphQL Indexer](https://docs.mintbase.xyz/dev/mintbase-graph) we can fetch the nfts from a specific smart contract - to filter by blog post we use 'blogpost' as an extra field as explained in the previous step.
+
+```typescript
+export const GET_BLOG_POSTS = `query GET_BLOG_POSTS($contractId: String!) {
+    mb_views_nft_tokens(
+      where: {extra: {_eq: "blogpost"}, _and: {nft_contract_id: {_eq: $contractId}}}
+    ) {
+      metadata_id
+      title
+      description
+      media
+      minted_timestamp
+    }
+  }
+  `;
 ```
-pnpm dev
+
+#### Get user blog posts (nfts)
+
+```typescript
+export const GET_USER_POSTS = `query GET_USER_POSTS($accountId: String!) {
+    mb_views_nft_tokens(
+      where: {extra: {_eq: "blogpost"}, _and: {nft_contract_owner_id: {_eq: $accountId}}}
+    ) {
+      metadata_id
+      title
+      description
+      media
+      minted_timestamp
+    }
+  }
+  `;
 ```
 
+#### Get user blogs (smart contracts)
+
+```typescript
+export const GET_USER_BLOGS = `query GET_USER_BLOGS($accountId: String!) {
+    nft_contracts(where: {owner_id: {_eq: $accountId}}) {
+      id
+    }
+  }
+  `;
+```
+
+#### Get latest blogs (smart contracts)
+
+```typescript
+export const GET_LATEST_UPDATED_BLOGS = `query GET_LATEST_UPDATED_BLOGS {
+  mb_views_nft_metadata(
+    where: {extra: {_eq: "blogpost"}}
+    distinct_on: nft_contract_id
+    limit: 10
+  ) {
+    nft_contract_id
+    nft_contract_owner_id
+  }
+  }
+  `;
+```
+
+#### Get latest blog posts (nfts)
+
+```typescript
+export const GET_LATEST_POSTS = `query GET_LATEST_POSTS {
+    mb_views_nft_tokens(
+      where: {extra: {_eq: "blogpost"}}
+      limit: 10
+      order_by: {minted_timestamp: desc}
+    ) {
+    metadata_id
+    title
+    description
+    media
+    minted_timestamp
+    minter
+    nft_contract_id
+    }
+  }`;
+```
+
+#### Get blog post (nft) data
+
+```typescript
+export const GET_POST_METADATA = `query GET_POST_METADATA($metadataId: String!) {
+    mb_views_nft_tokens(where: {metadata_id: {_eq: $metadataId}}) {
+      metadata_id
+      title
+      description
+      media
+      minted_timestamp
+      minter
+      nft_contract_id
+    }
+  }`;
+```
+
+Presently, this template exclusively functions within the testnet environment. To transition to a different network, it is imperative to modify the configuration in <MintbaseWalletContextProvider> and every 'testnet' instance.
```

Comment:
suggestion:` this works exclusively functions within testnet environment`

 `To change to a different network`